### DISCUSSION TITLE
set a background image for a room

### DISCUSSION THREAD

**Discussion by albjeremias** [element-hq/element-meta#625] [Ideas]
If I am a mod > 50, I should be able to setup a background image for a room :dancer:

**Comment by MTRNord** [comment]
related: vector-im/element-web#5844

**Comment by k80w** [comment]
I made a rough demo patch with a basic implementation of this [here](https://paste.sr.ht/%7Ekt/45e725690b3d532732086ddd1d1f2bcff97c7f63) ([raw](https://paste.sr.ht/blob/e8608fe29d0cf7d45affdaa9609215cbfa916b48)). The patch is ***extremely*** rough as it was only made to test the feature, and currently only works with the dark theme. Here's a screenshot
![screenshot](https://user-images.githubusercontent.com/4603019/56467547-57ec9700-63ee-11e9-919c-6b4926f729fd.png) 

My patch fills the background of the entire app with the image, rather than just the room's timeline. I personally found this more aesthetically pleasing, but if this were to be developed as a legitimate patch, there would likely need to be more discussion about where the image should actually be drawn.

My patch allows rooms to have a background set both in the rooms state as well as individual users' account room data, allowing users to override backgrounds for themselves. Again, discussion would need to be had to decide if this is necessary.

The patch defines a `host.katie.room.background` event which contains an image `url` and `fill` mode. `fill` basically just corresponds to the CSS `background-size` property, meaning users can set the image to either stretch to fill the screen or tile to fit it.

The JS for this patch is much simpler than I'd expected it to be, the biggest pain was the CSS; as my patch fills the entire background, it needed to make the backgrounds for many elements transparent. I doubt the CSS in the patch is up to the quality standards for this repo. There would likely be much less CSS hackiness if the background only applied to the timeline rather than the whole app.

**Comment by ThePharaohArt** [comment]
There should be a custom global background for your client.
And there should be an option in the client to allow rooms to override the background when in said room. Some people (me) would probably keep this overriding off.

**Reply by hoka99** [reply]
I think this should be how telegram works and a user can just set backgrounds for there specific client. I'm worried about admins having sooooo many options to choose from its overly complicated and daunting.
